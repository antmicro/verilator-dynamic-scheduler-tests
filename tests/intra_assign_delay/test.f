$TEST_DIR/test.sv --main --timing -Wno-INITIALDLY
