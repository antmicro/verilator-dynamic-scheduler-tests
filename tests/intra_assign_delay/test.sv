module t;
  int val;
  always @val $write("[%0t] val=%0d\n", $time, val);
  
  initial begin
    val = 1;
    #10 val = 2;
    fork #5 val = 4; join_none
    val = #10 val + 1;
    val <= #10 val + 2;
    fork #5 val = 6; join_none
    #20 $write("*-* All Finished *-*\n");
    $finish;
  end
endmodule
