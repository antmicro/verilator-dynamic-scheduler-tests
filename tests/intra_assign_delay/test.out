[0] val=1
[10] val=2
[15] val=4
[20] val=3
[25] val=6
[30] val=5
*-* All Finished *-*
